src/System.Data.SqlClient/tests/ManualTests/SQL/InstanceNameTest/InstanceNameTest.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Data.SqlClient.ManualTesting.Tests
+{
+    public static class InstanceNameTest
+    {
+        [CheckConnStrSetupFact]
+        public static void ConnectToSQLWithInstanceNameTest()
+        {
+            string tcpConnStr = Environment.GetEnvironmentVariable("TEST_TCP_CONN_STR");
+
+            string hostname = null;
+            string instanceName = null;
+            SqlConnectionStringBuilder builder = new SqlConnectionStringBuilder(tcpConnStr);
+            string[] dataSource = null;
+            if ((dataSource = builder.DataSource.Split('\\')).Length == 2)
+            {
+                hostname = dataSource[0].Replace("tcp:", "");
```

Comment:
The only time you're allowing the test to run is if the connection string already contains instance name info. If that's the case, why parse out this info at all? Why not just check that the string follows the correct format with a RegEx or something?